### ISSUE TITLE
When use the realtime database in Deno, it never finishes running

### ISSUE THREAD

**Issue by ayame113** [firebase/firebase-js-sdk#5783]
### [REQUIRED] Describe your environment

  * Operating System version: Microsoft Windows [Version 10.0.19042.1348]
  * Browser version: Deno 1.16.3
  * Firebase SDK version: 9.6.0
  * Firebase Product: realtime database

#### Steps to reproduce:

Running the following code with `deno run -A --location http://foo.com test.ts` will hang forever after the` finish` is outputted, even though I'm calling `deleteApp`.

```ts
// test.ts

// @deno-types="https://cdn.esm.sh/v58/firebase@9.6.0/app/dist/app/index.d.ts"
import { deleteApp, FirebaseOptions, initializeApp } from "https://www.gstatic.com/firebasejs/9.6.0/firebase-app.js";
// @deno-types="https://cdn.esm.sh/v58/firebase@9.6.0/database/dist/database/index.d.ts"
import { getDatabase, ref, set } from "https://www.gstatic.com/firebasejs/9.6.0/firebase-database.js";

const app = initializeApp({...});
const db = getDatabase(app);
await set(ref(db), 1);
deleteApp(app);
console.log("finish");
```

I changed it to log the setTimeout call using the following code and found that setTimeout called the `reportStats_` method forever.

```ts
const originalSetTimeout = globalThis.setTimeout;
globalThis.setTimeout = (...args) => {
  const timeoutId = originalSetTimeout(...args);
  console.log({ timeoutId }, ...args);
  return timeoutId;
};
```

The reason for this may be that the implementation of setTimeout in Deno is web compatible and there is no unref method.
We probably need to stop using the `unref` method and call `crearTimeout()` when `deleteApp()` is called.

https://github.com/firebase/firebase-js-sdk/blob/cdada6c68f9740d13dd6674bcb658e28e68253b6/packages/database/src/core/util/util.ts#L620-L625

**Comment by jbalidiong** [comment]
Hi @ayame113, thanks for the report. I was able to reproduce the behavior. Let me check what we can do for this issue or bring someone here that can provide more context about it. I’ll update this thread if I have any information to share.

**Comment by maneesht** [comment]
@ayame113 - we use `unref` to signal to the runtime that if the program wishes to exit earlier, do not wait on this `setTimeout` to finish. If we just clear the timeout right then and there, the stats will never get reported. 

I have an experimental PR created that uses Deno's `unrefTimer` as opposed to NodeJS's `unref`. It's experimental since the API that it requires `Deno.unrefTimer` is considered an "unstable" API, and in order for this to work, you need to run your app with `--unstable` to allow for unstable APIs to be used.

**Comment by maneesht** [comment]
Also, thank you for spending the time to root cause this issue. You made this very easy to debug and test.

**Comment by maneesht** [comment]
@ayame113 - I have another PR incoming to allow you to use Deno with local node environments, if you choose. But this issue has been resolved.

**Comment by ayame113** [comment]
> But this issue has been resolved.

nice!!
thank you!!
